Manage Logon Hour via Windows PowerShell

```diff
@@ -0,0 +1,345 @@
+
+![Manage%20Logon%20Hour%20via%20Windows%20PowerShell%20c28909d216f44708aad42cfa6706b403/Untitled%201.png](Manage%20Logon%20Hour%20via%20Windows%20PowerShell%20c28909d216f44708aad42cfa6706b403/Untitled%201.png)
+
+Logon Hour in attribute
+
+This is what represent the logon hours, but this is undoubtedly not a readable value and its not clear what these value means.
+
+## Understanding LogonHour Property
+
+In the **LogonHour**, each day of the week represented with 3 Byte (Three sets of **11111111** or **00000000**), and each 1 bit of the byte represent 1 hour
+
+![Manage%20Logon%20Hour%20via%20Windows%20PowerShell%20c28909d216f44708aad42cfa6706b403/Untitled%202.png](Manage%20Logon%20Hour%20via%20Windows%20PowerShell%20c28909d216f44708aad42cfa6706b403/Untitled%202.png)
+
+This user is allowed to login all the days though the week
+
+- One Hour = 1 Bit
+- Eight Hour = 1 Byte
+- One Day (24 Hour) = 3 Byte (24 Bit)
+- One Week = 21 Byte
+- One week in Hours= 168 bit, which also mean One week have 168 Hour
+
+All the result in the property are in a UTC standard.
+
+Changing any value from 1 to 0 or 0 to 1 will be reflected in the Logon Hours GUI. Let's try to change the first bit in the first block, make it 0, and then see the result. 
+
+So change the first block and make it **01111111**, click on OK to save, and open back the Logon Hour in the **Account** tab. See that there is only one block that is now marked as white.
+
+![Manage%20Logon%20Hour%20via%20Windows%20PowerShell%20c28909d216f44708aad42cfa6706b403/Untitled%203.png](Manage%20Logon%20Hour%20via%20Windows%20PowerShell%20c28909d216f44708aad42cfa6706b403/Untitled%203.png)
+
+Result After Changing 1 bit
+
+In this post, what you see might "*till this point*" not look like what you see on your side, even after doing the same change, because the **LogonHour** property uses UTC standard time format.
+
+I will explain more about this issue and how to fix it via the PowerShell script.
+
+## Parameter for the PowerShell Script.
+
+To make things easy to use for the user, I wrote a function **Set-LogonHours** that accepts the following parameter.
+
+- **Identity:** The Username of the target principal (*String, Accept Pipeline, Position=0)*
+- **TimeIn24Format:** Array of the hours access is permitted (*Array, ValidateRange (0- 23)*
+- **Sunday:** Apply the TimeIn24Format to this day *(Switch)*
+- **Monday:** Apply the TimeIn24Format to this day *(Switch)*
+- **Tuesday:** Apply the TimeIn24Format to this day *(Switch)*
+- **Wednesday:** Apply the TimeIn24Format to this day *(Switch)*
+- **Thursday:** Apply the TimeIn24Format to this day *(Switch)*
+- **Friday:** Apply the TimeIn24Format to this day *(Switch)*
+- **Saturday:** Apply the TimeIn24Format to this day *(Switch)*
+- **NonSelectedDaysAre:** This option for the weekdays that are not selected, what the default value should be, is it a permit or denied *(ValidateSet ="WorkingDays" or "NonWorkingDays").*
```

Comment:
For this blog, ignore all the accept pipeline stuff. Not needed for this article